src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -2492,7 +2492,7 @@ private void ClassifyUpdate(VariableDeclaratorSyntax oldNode, VariableDeclarator
                 }
 
                 // Check if a constant field is updated:
-                var fieldDeclaration = (FieldDeclarationSyntax)oldNode.Parent.Parent;
+                var fieldDeclaration = (BaseFieldDeclarationSyntax)oldNode.Parent.Parent;
```

Comment:
even this seems a bit unsafe to me.  I'm not seeing anythin obvious above how we know this is parented by somethin field-like.   could this be an `is/as` instead?

Also, let's say we know this is in top-level declaration context... this still feels unsafe since we might add language features in the future that involve a VariableDeclaratorSyntax in some new location.  So checking the actual shape feels appropriate IMO.

I'll leave to @tmat to state what he thinks the best strategy is here.

Reply:
The code needs to handle all possible cases exhaustively, so there is really nothing we can do if this is not `BaseFieldDeclarationSyntax`.

The code needs to be kept in sync with latest language features. It is better if it throws than if the error was swallowed and the EnC produced IL/metadata delta that corrupts the app. 

Note that any exception thrown by this code is recoverable and won't cause VS to crash, but it will send a non-fatal Watson to us.

Reply:
`is` seems to be safer. Thank you, @CyrusNajmabadi and @tmat ! Sent an update.